darts/models/transformer_model.py

```diff
@@ -0,0 +1,281 @@
+                 nhead: int = 8,
+                 num_encoder_layers: int = 6,
+                 num_decoder_layers: int = 6,
+                 dim_feedforward: int = 2048,
+                 dropout: float = 0.1,
+                 activation: str = "relu",
+                 custom_encoder: Optional[Any] = None,
+                 custom_decoder: Optional[Any] = None,
+                 random_state: Optional[Union[int, RandomState]] = None,
+                 **kwargs):
+
+        """
+        Transformer is a state-of-the-art deep learning model introduced in 2017. It is an encoder-decoder
+        architecture whose core feature is the 'multi-head attention' mechanism, which is able to
+        draw intra-dependencies within the input vector and within the output vector ('self-attention')
+        as well as inter-dependencies between input and output vectors ('encoder-decoder attention').
+        The multi-head attention mechanism is highly parallelizable, which makes the transformer architecture
+        very suitable to be trained with GPUs.
+
+
+        The transformer architecture implemented here is based on the paper “Attention Is All You Need”:
+        Ashish Vaswani, Noam Shazeer, Niki Parmar, Jakob Uszkoreit, Llion Jones, Aidan N Gomez, Lukasz Kaiser, and
+        Illia Polosukhin. 2017. Attention is all you need. In Advances in Neural Information Processing Systems,
+        pages 6000-6010.
+        (paper can be found at https://arxiv.org/abs/1706.03762)
+
+        Parameters
+        ----------
+        model
+            a custom PyTorch module with the same specifications as
+            `darts.models.transformer_model._TransformerModule` (default=None).
+        input_size
+            The dimensionality of the TimeSeries instances that will be fed to the fit function (default=1).
+        input_length
+            Number of time steps to be output by the forecasting module (default=1).
+        output_size
+            The dimensionality of the output time series (default=1).
+        output_length
+            Number of time steps to be output by the forecasting module (default=1).
+        d_model
+            the number of expected features in the transformer encoder/decoder inputs (default=512).
+        nhead
+            the number of heads in the multiheadattention model (default=8).
+        num_encoder_layers
+            the number of encoder layers in the encoder (default=6).
+        num_decoder_layers
+            the number of decoder layers in the decoder (default=6).
+        dim_feedforward
+            the dimension of the feedforward network model (default=2048).
+        dropout
+            Fraction of neurons affected by Dropout (default=0.1).
+        activation
+            the activation function of encoder/decoder intermediate layer, 'relu' or 'gelu' (default='relu').
+        custom_encoder
+            a custom transformer encoder provided by the user (default=None)
```

Comment:
I would say "a custom user-provided encoder module for the transformer"
(this is because "provided by the user" sounds weird, as the user will likely be the person reading this docstring).